src/pe/export.rs

```diff
@@ -157,23 +157,21 @@ impl<'a> scroll::ctx::TryFromCtx<'a, scroll::Endian> for Reexport<'a> {
             let c: u8 = bytes.pread(o)?;
             debug!("reexport offset: {:#x} char: {:#x}", o, c);
             if c == b'.' {
-                let i = o - 1;
-                let dll: &'a str = bytes.pread_with(0, scroll::ctx::StrCtx::Length(i))?;
+                let dll: &'a str = bytes.pread_with(0, scroll::ctx::StrCtx::Length(o))?;
                 debug!("dll: {:?}", &dll);
-                let len = reexport_len - i - 1;
-                let rest: &'a [u8] = bytes.pread_with(o, len)?;
+                if o + 1 == reexport_len {
+                    break;
+                }
+                let len = reexport_len - o - 1;
+                let rest: &'a [u8] = bytes.pread_with(o + 1, len)?;
                 debug!("rest: {:?}", &rest);
-                let len = rest.len() - 1;
                 if rest[0] == b'#' {
-                    // UNTESTED
-                    let ordinal = rest.pread_with::<&str>(1, scroll::ctx::StrCtx::Length(len))?;
+                    let ordinal = rest.pread_with::<&str>(1, scroll::ctx::StrCtx::Length(len - 1))?;
```

Comment:
Have you tested this? It doesn't seem correct to me. `rest` is already starting at `o + 1`, so I don't think we need the additional `1` offset here.

Reply:
I did test. I changed `rest` to start after the `.` - so that's `o + 1`.

In the ordinal case, we also need to skip the `#` (start at offset 1), and the len includes the `#`, so subtract 1.

```
#175 PS> .\target\debug\pedump.exe C:\windows\System32\certcli.dll | sls mySan

Export { name: Some("mySanitizeName"), offset: 420153, rva: 423225, size: 0, reexport: Some(DLLOrdinal { export: "CERTCA", ordinal: 831 }) }
Export { name: Some("mySanitizedNameToDSName"), offset: 420189, rva: 423261, size: 0, reexport: Some(DLLOrdinal { export: "CERTCA", ordinal: 832 }) }
Export { name: Some("mySanitizedNameToShortName"), offset: 420228, rva: 423300, size: 0, reexport: Some(DLLOrdinal { export: "CERTCA", ordinal: 833 }) }

#176 PS> link /dump /exports C:\windows\System32\certcli.dll | sls mySan

    349   A0          mySanitizeName (forwarded to CERTCA.#831)
    350   A1          mySanitizedNameToDSName (forwarded to CERTCA.#832)
    351   A2          mySanitizedNameToShortName (forwarded to CERTCA.#833)
```

Reply:
Ah great, my mistake.